pkg/cmd/repo/create/http.go

```diff
@@ -10,79 +10,102 @@ import (
 	"github.com/cli/cli/api"
 )
 
-// repoCreateInput represents input parameters for repoCreate
 type repoCreateInput struct {
-	Name        string `json:"name"`
-	Visibility  string `json:"visibility"`
-	HomepageURL string `json:"homepageUrl,omitempty"`
-	Description string `json:"description,omitempty"`
-
-	OwnerID string `json:"ownerId,omitempty"`
-	TeamID  string `json:"teamId,omitempty"`
+	Name                 string
+	HomepageURL          string
+	Description          string
+	Visibility           string
+	OwnerLogin           string
+	TeamSlug             string
+	TemplateRepositoryID string
+	HasIssuesEnabled     bool
+	HasWikiEnabled       bool
+	GitIgnoreTemplate    string
+	LicenseTemplate      string
+}
 
-	HasIssuesEnabled  bool   `json:"hasIssuesEnabled"`
-	HasWikiEnabled    bool   `json:"hasWikiEnabled"`
+type createRepositoryInputV3 struct {
+	Name              string `json:"name"`
+	HomepageURL       string `json:"homepage,omitempty"`
+	Description       string `json:"description,omitempty"`
+	IsPrivate         bool   `json:"private"`
+	Visibility        string `json:"visibility,omitempty"`
+	TeamID            uint64 `json:"team_id,omitempty"`
+	HasIssuesEnabled  bool   `json:"has_issues"`
+	HasWikiEnabled    bool   `json:"has_wiki"`
 	GitIgnoreTemplate string `json:"gitignore_template,omitempty"`
 	LicenseTemplate   string `json:"license_template,omitempty"`
 }
 
-type repoTemplateInput struct {
-	Name       string `json:"name"`
-	Visibility string `json:"visibility"`
-	OwnerID    string `json:"ownerId,omitempty"`
+type createRepositoryInput struct {
```

Comment:
It isn't immediately obvious from the name what the difference between `repoCreateInput`, `createRepositoryInputV3`, and `createRepositoryInput` is, I think comments about their purpose could help clear that up.